kratos/tests/test_skin_detection_process.py

```diff
@@ -122,10 +96,74 @@ def test_NotOnSubModelPartSkinDetectionProcess(self):
+        else:
+            self.assertEqual(self.model_part.NumberOfConditions(), 112)
+
+    def _post_process(self, model_part):
+        gid_output = GiDOutputProcess(model_part,
+                                    "gid_output",
+                                    KratosMultiphysics.Parameters("""
+                                        {
+                                            "result_file_configuration" : {
+                                                "gidpost_flags": {
+                                                    "GiDPostMode": "GiD_PostBinary",
+                                                    "WriteDeformedMeshFlag": "WriteUndeformed",
+                                                    "WriteConditionsFlag": "WriteConditions",
+                                                    "MultiFileFlag": "SingleFile"
+                                                },
+                                                "nodal_flags_results" : ["INTERFACE","ACTIVE"]
+                                            }
+                                        }
+                                        """)
+                                    )
+
+        gid_output.ExecuteInitialize()
+        gid_output.ExecuteBeforeSolutionLoop()
+        gid_output.ExecuteInitializeSolutionStep()
+        gid_output.PrintOutput()
+        gid_output.ExecuteFinalizeSolutionStep()
+        gid_output.ExecuteFinalize()
+
+    def _post_process_mpi(self, model_part):
+        data_comm = model_part.GetCommunicator().GetDataCommunicator()
+        rank = data_comm.Rank()
+        gid_output = GiDOutputProcess(model_part,
+                                    "gid_output_"+str(rank),
+                                    KratosMultiphysics.Parameters("""
+                                        {
+                                            "result_file_configuration" : {
+                                                "gidpost_flags": {
+                                                    "GiDPostMode": "GiD_PostBinary",
+                                                    "WriteDeformedMeshFlag": "WriteUndeformed",
+                                                    "WriteConditionsFlag": "WriteConditions",
+                                                    "MultiFileFlag": "SingleFile"
+                                                },
+                                                "nodal_results": ["PARTITION_INDEX"],
+                                                "nodal_flags_results" : ["INTERFACE","ACTIVE"]
+                                            }
+                                        }
+                                        """)
+                                    )
+
+        gid_output.ExecuteInitialize()
+        gid_output.ExecuteBeforeSolutionLoop()
+        gid_output.ExecuteInitializeSolutionStep()
+        gid_output.PrintOutput()
+        gid_output.ExecuteFinalizeSolutionStep()
+        gid_output.ExecuteFinalize()
```

Comment:
why not using the respective python processes to avoid code duplication

or even better, remove the post stuff altogether?

Reply:
But but is useful...